src/datachain/asyn.py

```diff
@@ -224,3 +224,23 @@ async def _end_iteration(self) -> None:
     async def _break_iteration(self) -> None:
         self.heap = []
         self._push_result(self._next_yield, None)
+
+
+def iter_over_async(ait, loop):
```

Comment:
[Q] Should this handle exceptions in the same way as `AsyncMapper.iterate`? Is it possible/worth the effort to refactor and have `AsyncMapper.iterate` call this function?

Reply:
I don't think it makes sense to refactor `AsyncMapper.iterate()` with this method as `AsyncMapper` is implemented for different scenario where we know in advance how many tasks we expect so it's `.iterate()` method doesn't really deal with `AsyncIterator` .   
When doing bucket listing we don't know in advance how many async tasks / futures will be there as we don't know the depth of the directory tree so, as a result, we get `AsyncIterator` and this function just does simple conversion to the sync one.

Regarding exception handling, I'm not sure about that  either. I've added a test with exception and it works fine. From what I see in `AsyncMapper` we explicitly run through futures / tasks to check for exceptions but I don't see the need to do so here. I mean, I could maybe do:
```python
future = asyncio.run_coroutine_threadsafe(get_next(), loop)
if err := future.exception():
    raise err
done, obj = future.result()
...
```
but it seems to work without that as well (which is expected IMO)